### PR TITLE
Generate docs when using library mode

### PR DESCRIPTION
**Pull request by skeet70**
Adds a call to generate docs when using library mode (as recommended by the docs for proc macros). This worked in our medium sized project that uses only proc-macros+library mode.

### CODE REVIEW

**Comment by MeerKatDev** [comment]
Hi, could you solve the conflicts? I'm sorry if I messed up with them myself. Let me know if I can somehow help.

**Comment by skeet70** [comment]
> Hi, could you solve the conflicts? I'm sorry if I messed up with them myself. Let me know if I can somehow help.

I merged main in. It's still working for our project and the tests I can run still pass.

File: uniffi_bindgen/src/library_mode.rs [hunk 1]
```diff
@@ -100,7 +100,12 @@ pub fn generate_external_bindings<T: BindingGenerator>(
         }
     }
 
-    for source in sources.iter() {
+    for source in sources.iter_mut() {
```
[the remark this answers is not in the document]
Comment (MeerKatDev) [reply uniffi_bindgen/src/library_mode.rs:103]: Also, in the last version of the code, this part was extracted

File: uniffi_bindgen/src/bindings/swift/templates/ErrorTemplate.swift [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
     {% if e.is_flat() %}
     {% for variant in e.variants() %}
+    {% include "EnumVariantDocsTemplate.swift" %}
     // Simple error enums only carry a message
     case {{ variant.name()|class_name }}(message: String)
     {% endfor %}
 
     {%- else %}
     {% for variant in e.variants() %}
+    {% include "EnumVariantDocsTemplate.swift" %}
```
Comment (MeerKatDev) [inline uniffi_bindgen/src/bindings/swift/templates/ErrorTemplate.swift:13]: thank you!

**Review by MeerKatDev — CHANGES_REQUESTED** [review]

File: uniffi_bindgen/src/library_mode.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MeerKatDev) [reply uniffi_bindgen/src/library_mode.rs:103]: I mean - is it possible to reuse the code here?

[the remark this answers is not in the document]
Comment (skeet70) [reply uniffi_bindgen/src/library_mode.rs:103]: Very possible, I've just been blazing forward with using this, trying to release and document the product we're building with it. I have another hackday in two weeks, but I can try to come back and take a look at consolidating in between tickets before then.

[the remark this answers is not in the document]
Comment (skeet70) [reply uniffi_bindgen/src/library_mode.rs]: I think I've done this, and removed the methods I had added to do it. If you were thinking of a different reusable part let me know.

**Pull request closed without merging**